src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,124 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_INITIAL_CAPACITY = 1 << 4;
+    static final int MAXIMUM_CAPACITY = 1 << 30;
+    static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int size;
+    private Node<K,V>[] table;
+    private int threshold;
+    private int capacity;
+    private float loadFactor;
 
     @Override
     public void put(K key, V value) {
-
+        int hash = hash(key);
+        Node<K,V> currentNode;
+        Node<K,V> prev;
+        if (table == null || table.length == 0) {
+            resize();
+        }
+        int position = hash % table.length;
+        K currKey;
+        if ((currentNode = table[position]) == null) {
+            table[position] = new Node<>(hash, key, value, null);
+        } else {
+            do {
+                if (currentNode.hash == hash && ((currKey = currentNode.key) == key
+                        || (key != null && key.equals(currKey)))) {
+                    currentNode.value = value;
+                    return;
+                }
+                prev = currentNode;
+            } while ((currentNode = currentNode.next) != null);
+            prev.next = new Node<>(hash, key, value, null);
+        }
+        if (++size > threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int hash = hash(key);
+        Node<K,V> currentNode;
+        K keyCurrent;
+        if (table != null && table.length > 0
+                && (currentNode = table[hash % table.length]) != null) {
+            do {
+                if (currentNode.hash == hash && ((keyCurrent = currentNode.key) == key
+                        || (key != null && key.equals(keyCurrent)))) {
+                    return currentNode.value;
+                }
+            } while ((currentNode = currentNode.next) != null);
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private static int hash(Object key) {
+        int h;
+        return (key == null) ? 0 : (((h = key.hashCode()) < 0) ? h * -1 : h);
```

Comment:
The use of `Math.abs` in the hash function can lead to negative values when dealing with `Integer.MIN_VALUE`. Consider using a different method to ensure that hash values are always non-negative.